7_react/react1/week2/todo-app/src/components/Timer/Timer.jsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+
+
+
+const Timer = () => {
+
+    const [seconds, setSeconds] = useState(0);
+    const [minutes, setMinutes] = useState(0);
+    
+    useEffect(()=>{
+        const timer = setInterval(()=>{
+            setSeconds(prevSeconds => {
```

Comment:
Awesome work with the split into minutes and seconds! 🎉 Much more appealing to look at.

There is a small issue with the updater function for seconds: it is not pure! It has a side-effect since it calls `setMinutes`. You might have seen when running in development mode, that your minute counter increments twice every time 60 seconds pass. This is due to it running it strict mode where it runs functions supposed to be pure twice to provoke purity issues during development. [You can read more here.](https://react.dev/reference/react/useState#my-initializer-or-updater-function-runs-twice)

Can you think of a way to do this with only one simple state and not two? Then it becomes a lot easier to manager. 😄

Reply:
I created one state `elapsedTime`. Can you check from my commits:`Fix Timer bug`? Thank you for helping with this bug🙏